### ISSUE TITLE
[bug] S3 bucket for terraform remote state file got deleted before destroy completed

### ISSUE THREAD

**Issue by chunyong-lin** [airbnb/streamalert#1048]
## Background
When deploy StreamAlert `release-3-0-0` branch to stage environment, I found that the S3 bucket for terraform remote state file got deleted before destroy completed, so that terraform throws error message
```
Failed to save state: failed to upload state: NoSuchBucket: The specified bucket does not exist
	status code: 404 ...
```

### Steps to Reproduce
```
python manage.py destroy
```
## Desired Change
Either delete S3 bucket for terraform backend in the end of resource destroy or terraform only uses local state file and not try to read remote state file again during destroy.

**Comment by jack1902** [comment]
Got this issue today, trying to clean up a test environment. Ideally we should switch from remote to local backend and then run destroy? (this is the behaviour from 2.0?)

**Comment by chunyong-lin** [comment]
Fixed in PR #1093
